README.md

```diff
@@ -460,6 +467,22 @@ is set to `'true'`, it will be used as header value as is, without any extra mod
 | gid.connector.http.sink.writer.request.mode             | optional | Sets Http Sink request submission mode. Two modes are available to select, `single` and `batch` which is the default mode if option is not specified.                                                                                            |
 | gid.connector.http.sink.request.batch.size              | optional | Applicable only for `gid.connector.http.sink.writer.request.mode = batch`. Sets number of individual events/requests that will be submitted as one HTTP request by HTTP sink. The default value is 500 which is same as HTTP Sink `maxBatchSize` |
 
+
+## Lookup Cache 
+The HTTP Client connector can be used in temporal join as a lookup source (aka. dimension table).  
+
+By default, lookup cache is not enabled. You can enable it by setting `lookup.cache` to `PARTIAL`. Caching is only enabled if `asyncPolling` = false.
+The scope of the cache is for per job, so long running jobs can benefit from this caching.
+
+The lookup cache is used to improve performance of temporal join the HTTP Client connector. By default, lookup cache is not enabled, so all the API requests are sent on the network. When lookup cache is enabled, Flink will look in the cache first, and only send requests 
+on the network when there is no cached value, and update cache with the rows returned. The oldest rows in cache will be expired when the cache hits the max cached rows `lookup.partial-cache.max-rows` or when the row exceeds the max time to live specified by `lookup.partial-cache.expire-after-write` or `lookup.partial-cache.expire-after-access`.
+The cached rows might not be the latest, users can tune expiration options to a smaller value to have a better fresh data, but this may increase the number of API requests sent. So this is a balance between throughput and correctness.
```

Comment:
(minor)
```suggestion
The cached rows might not be the latest. Users can tune expiration options to a smaller value to have fresher data, but this may increase the number of sent API requests. So this is a balance between throughput and correctness.
```